functions/index.js

```diff
@@ -12,6 +13,32 @@ admin.initializeApp(functions.config().firebase);
 exports.createNewUser = functions.auth.user().onCreate(event => {
   const uid = event.data.uid;
   const email = event.data.email;
-  
+
   return admin.database().ref(`/users/${uid}`).set({ email });
 });
+
+exports.joinTheClub = functions.database.ref('/queues/joinTheClub/tasks/{taskId}').onWrite(event => {
+  if (event.data._newData) {
+    const { inviteCode, userId } = event.data._newData;
+
+    return admin.database().ref('bookClub').orderByChild('inviteCode').equalTo(inviteCode).once('value').then((response) => {
+      if (response.exists() && response.val()) {
+        const val = response.val();
+        const bookClubId = Object.keys(val)[0];
+        const bookClubName = Object.keys(val)[0].name;
+        const randomId = Math.random().toString(36).slice(-10);
+
+        const updates = {
+          [`bookClub/${bookClubId}/users/${userId}`]: true,
+          [`users/${userId}/bookClubs/${bookClubId}`]: true,
+          [`queues/joinTheClub/tasks/${event.params.taskId}`]: null,
+          [`users/${userId}/notifications/${randomId}`]: { message: `You were successfully added to ${bookClubName}` }
+        };
+        return admin.database().ref().update(updates);
+      } else {
+        admin.database().ref('queues/joinTheClub/tasks').child(event.params.taskId).remove();
+        return admin.database().ref(`users/${userId}/notifications`).push({ message: 'The invite code you submitted didn\'t work' });
```

Comment:
When you do `.push()`, firebase generates one for you. In the other scenario I was doing a giant update and it wouldn't have worked. If something goes wrong with part of the update, firebase will revert what it changed so we don't have bad data, hence the giant update